kubespawner/spawner.py

```diff
@@ -2409,44 +2397,64 @@ def _start_reflector(
         If replace=True, a running pod reflector will be stopped
         and a new one started (for recovering from possible errors).
         """
-        key = kind
-        ReflectorClass = reflector_class
 
-        def on_reflector_failure():
-            self.log.critical(
-                "%s reflector failed, halting Hub.",
-                key.title(),
-            )
-            sys.exit(1)
+        if self.enable_user_namespaces:
+            # One reflector for all namespaces (requires ClusterRole).
+            # Stop JupyterHub if something went wrong
 
-        previous_reflector = self.__class__.reflectors.get(key)
+            def on_reflector_failure():
+                self.log.critical(
+                    "%s reflector failed, halting Hub.",
+                    key.title(),
+                )
```

Comment:
Since `key` is a tuple, title() will error.

```suggestion
                self.log.critical(f"{key} reflector failed, halting Hub.")
```

Reply:
Using f-strings in logs is not recommended. Intergrations like Sentry or Elastic APM are using template strings to count number of errors caused by a specific issue, but f-strings produces different strings for each substitution values combination

Reply:
Makes sense, I updated the code suggestion and mentioned your thought in an existing discussion at https://github.com/jupyterhub/team-compass/issues/625!